src/ai/backend/client/cli/params.py

```diff
@@ -166,3 +168,14 @@ def convert(self, arg, param, ctx):
                 return arg.split(",")
         except ValueError as e:
             self.fail(repr(e), param, ctx)
+
+
+class OptionalType(click.ParamType):
+    name = "Optional Type Wrapper"
+
+    def convert(self, value, param, ctx):
+        try:
+            if isinstance(value, (str, int)) or value == undefined:
+                return value
+        except ValueError as e:
+            self.fail(repr(e), param, ctx)
```

Comment:
```suggestion
    def __init__(self, type_: type) -> None:
        super().__init__()
        self.type_ = type_

    def convert(self, value: Any, param, ctx):
        if isinstance(value, self.type_) or isinstance(value, undefined):
            return value
        self.fail(f"{value!r} is not valid `{self.type_}` or `undefined`", param, ctx)
```

Let `OptionalType` handle not only `str` and `int` types but also Any given type.
And it is good to pass detail failure messages.
Then, `OptionalType` can be used like below
```python
# example
@click.option("--name", type=OptionalType(str), default=undefined, help="Example option.")
```